### PR TITLE
Feature/toolboxes refactoring

### PR DESCRIPTION
**Pull request by vincentchabannes**
some updates from toolboxes refactoring : 

 - New feature in Json sections : Parameters, Materials,PostProcess
 - Update heat toolbox documentation as a reference template for others (should be updated)

### CODE REVIEW

File: docs/toolboxes/modules/ROOT/pages/modeling-analysis-using-json-files.adoc [hunk 1]
```diff
@@ -42,36 +42,40 @@ Here is an example of this section for the `fluid` toolbox.
...
-<3> defines a new parameter `chi` and its associated expression (currently symbols supported are `x`, `y`, `z`, `nx`, `ny`, `nz`, `t`
-<4> defines a new parameter `pIn` and its definition is given in the subsection below 
-<5> the type of parameter is fit
-<6> the filename of a csv file used for the fitting
-<7> column name of csv file used in abscissa
-<8> column name of csv file used in ordinate
-<9> interpolation type of the fit. Possible values are : `P0`, `P1`, `Spline`, `Akima`
-<10> expression used in order to read the fitted value
+<3> defines a new parameter `alpha` and its associated expression. This expression depends on another symbol, here the parameter `ubar`. The symbol defined by this new parameter is also called `alpha`.
+<4> defines a new parameter `beta` and its associated expression. Here the expression is vector of dimension 2. Consequently, symbols generated by this new parameter are `alpha_x` and `alpah_y` (Currently we cannot use a vector as a symbol).
```
Comment (romainhild) [inline docs/toolboxes/modules/ROOT/pages/modeling-analysis-using-json-files.adoc]:
```suggestion
<4> defines a new parameter `beta` and its associated expression. Here the expression is vector of dimension 2. Consequently, symbols generated by this new parameter are `beta_x` and `beta_y` (Currently we cannot use a vector as a symbol).
```

File: docs/toolboxes/modules/ROOT/pages/modeling-analysis-using-json-files.adoc [hunk 2]
```diff
@@ -290,8 +290,55 @@ This section allows to define the output fields and quantities to be computed an
...
+         "expr":"{materials_k_xx,materials_k_xy,materials_k_yx,materials_k_yy}:materials_k_xx:materials_k_xy:materials_k_yx:materials_k_yy", <12>
+         "representation":["nodal","element"]
+      }
+   }
+----
+<1> exports fields that are available in the toolbox used (see the toolbox documentation).
+<2> start the expression subsection
+<3> export a field named `toto` from a mathematical expression defined on the whole mesh
+<4> export a field named `titi` from mathematical expressions
+<5> start a section named `part` in order to tell that the exported fields is defined from several expressions related to a part of the mesh
```
Comment (romainhild) [inline docs/toolboxes/modules/ROOT/pages/modeling-analysis-using-json-files.adoc:370]:
```suggestion
<5> start a section named `parts` in order to tell that the exported fields is defined from several expressions related to a part of the mesh
```

File: docs/toolboxes/modules/ROOT/pages/modeling-analysis-using-json-files.adoc (hunk 1 above)
Comment (vincentchabannes) [reply docs/toolboxes/modules/ROOT/pages/modeling-analysis-using-json-files.adoc]: @romainhild I don't understand, my version is not the same : 
`
 <4> defines a new parameter `beta` and its associated expression. Here the expression is vector of dimension 2. Consequently, symbols generated by this new parameter are `beta_0` and `beta_1` (Currently we cannot use a vector as a symbol).
`

Comment (romainhild) [reply docs/toolboxes/modules/ROOT/pages/modeling-analysis-using-json-files.adoc]: ah maybe I made the comment on a previous commit, sorry

**Pull request merged**